src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,135 @@
+import Poller from '../poller'
+import type { Options } from '../types'
+import { getRandom } from '../utils'
+
+jest.useFakeTimers()
+
+jest.mock('../utils.ts', () => ({
+  getRandom: jest.fn()
+}))
+
+interface PollerArgs {
+  fetchFlags: jest.MockedFunction<() => Promise<any>>
+  configurations: Partial<Options>
+  pollInterval: number
+  maxAttempts: number
+}
+
+interface TestArgs {
+  delayFunction: jest.MockedFunction<() => number>
+  delayMs: number
+}
+
+const getPoller = (overrides: Partial<PollerArgs> = {}): { poller: Poller; pollerArgs: PollerArgs } => {
+  const args: PollerArgs = {
+    fetchFlags: jest.fn(),
+    configurations: {},
+    pollInterval: 60000,
+    maxAttempts: 5,
+    ...overrides
+  }
+
+  return {
+    poller: new Poller(args.fetchFlags, args.configurations, args.pollInterval),
+    pollerArgs: args
+  }
+}
+
+const getTestArgs = (overrides: Partial<TestArgs> = {}): TestArgs => {
+  const { delayMs = 5000 } = overrides // Extract delayMs or set a default of 5000
+
+  return {
+    delayMs,
+    delayFunction: (getRandom as jest.Mock).mockReturnValue(delayMs),
+    ...overrides
+  }
+}
+
+describe('Poller', () => {
+  it('should not start polling if it is already polling', () => {
+    const { poller } = getPoller()
+    const logSpy = jest.spyOn(poller as any, 'logDebug')
+
+    poller.start()
+    poller.start()
+
+    expect(logSpy).toHaveBeenCalledWith(
+      'Polling was requested but is already running - only one poller can run at a time.'
+    )
+  })
+
+  it('should retry fetching if there is an error', async () => {
+    const { poller, pollerArgs } = getPoller()
+    const fetchFlags = pollerArgs.fetchFlags
+    fetchFlags.mockResolvedValue(new Error('Fetch Error'))
+
+    const testArgs = getTestArgs()
+
+
+    const logSpy = jest.spyOn(poller as any, 'logDebug')
+
+    poller.start()
+
+    jest.advanceTimersByTime(pollerArgs.pollInterval)
```

Comment:
This is kinda pointless since `pollInterval` isn't actually used in the code apart from a debug message. `configuration.pollingInterval` is actually used by the code and isn't set, so it'll actually be calling `setInterval(callback, undefined)` and instantly returning.